### PR TITLE
Do not query vertex data in K_PATHS queries if vertex data is not needed

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/17547
Enterprise companion PR: https://github.com/arangodb/enterprise/pull/1139

Don't query vertex data in K_PATHS cluster implementation in case the query doesn't need them.
This is an experimental change and needs more testing! Do not yet use in production.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.10: this PR
  - [x] Backport for 3.9: https://github.com/arangodb/arangodb/pull/17521
  - [ ] Backport for 3.8: -

#### Related Information

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/1139
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Graph/Enumerators/OneSidedEnumerator.cpp [hunk 1]
```diff
@@ -121,11 +121,11 @@ auto OneSidedEnumerator<Configuration>::computeNeighbourhoodOfNextVertex()
 
     // Will throw all network errors here
     std::vector<Step*> preparedEnds = std::move(futureEnds.get());
-
     TRI_ASSERT(preparedEnds.size() != 0);
     TRI_ASSERT(_queue.firstIsVertexFetched());
   }
 
+  TRI_ASSERT(!_queue.isEmpty());
```
Comment (hkernbach) [inline arangod/Graph/Enumerators/OneSidedEnumerator.cpp:128]: This is not required to be here I guess, as we're checking this already in Line 116. But it does not harm.

File: arangod/Graph/Providers/ClusterProvider.cpp [hunk 2]
```diff
@@ -130,28 +130,49 @@ auto ClusterProvider<StepImpl>::startVertex(const VertexType& vertex,
...
   auto* pool =
       trx()->vocbase().server().template getFeature<NetworkFeature>().pool();
 
   network::RequestOptions reqOpts;
   reqOpts.database = trx()->vocbase().name();
   reqOpts.skipScheduler = true;  // hack to avoid scheduler queue
 
+  auto const* engines = _opts.engines();
   std::vector<Future<network::Response>> futures;
   futures.reserve(engines->size());
```
Comment (hkernbach) [inline arangod/Graph/Providers/ClusterProvider.cpp:177]: Same suggestion already mentioned here: https://github.com/arangodb/arangodb/pull/17547#discussion_r1019008799

**Review by hkernbach — APPROVED** [review]
lgtm, only two minor comments.

File: arangod/Graph/Enumerators/OneSidedEnumerator.cpp (hunk 1 above)
Comment (jsteemann) [inline arangod/Graph/Enumerators/OneSidedEnumerator.cpp:128]:
```suggestion
```

**Review by markuspf — APPROVED** [review]

File: arangod/Graph/Providers/ClusterProvider.cpp (hunk 2 above)
Comment (jsteemann) [reply arangod/Graph/Providers/ClusterProvider.cpp:177]: The `engines` variables is also used further down, so we cannot remove it.

File: arangod/Graph/Enumerators/OneSidedEnumerator.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Graph/Enumerators/OneSidedEnumerator.cpp]: Done

**Pull request merged**